diff --git a/node_modules/expo-modules-core/android/src/main/java/expo/modules/interfaces/barcodescanner/BarCodeScannerInterface.kt b/node_modules/expo-modules-core/android/src/main/java/expo/modules/interfaces/barcodescanner/BarCodeScannerInterface.kt
new file mode 100644
index 0000000..63f5db1
--- /dev/null
+++ b/node_modules/expo-modules-core/android/src/main/java/expo/modules/interfaces/barcodescanner/BarCodeScannerInterface.kt
@@ -0,0 +1,11 @@
+package expo.modules.interfaces.barcodescanner
+
+import android.graphics.Bitmap
+
+interface BarCodeScannerInterface {
+  val isAvailable: Boolean
+  fun scan(imageData: ByteArray, width: Int, height: Int, rotation: Int): BarCodeScannerResult?
+  fun scanMultiple(bitmap: Bitmap): List<BarCodeScannerResult>
+  fun setSettings(settings: BarCodeScannerSettings)
+}
+
diff --git a/node_modules/expo-modules-core/android/src/main/java/expo/modules/interfaces/barcodescanner/BarCodeScannerProviderInterface.kt b/node_modules/expo-modules-core/android/src/main/java/expo/modules/interfaces/barcodescanner/BarCodeScannerProviderInterface.kt
new file mode 100644
index 0000000..3cf1ac5
--- /dev/null
+++ b/node_modules/expo-modules-core/android/src/main/java/expo/modules/interfaces/barcodescanner/BarCodeScannerProviderInterface.kt
@@ -0,0 +1,9 @@
+package expo.modules.interfaces.barcodescanner
+
+import android.content.Context
+
+interface BarCodeScannerProviderInterface {
+  fun createBarCodeScanner(context: Context): BarCodeScannerInterface?
+  fun createBarCodeDetectorWithContext(context: Context): BarCodeScannerInterface
+}
+
diff --git a/node_modules/expo-modules-core/android/src/main/java/expo/modules/interfaces/barcodescanner/BarCodeScannerResult.kt b/node_modules/expo-modules-core/android/src/main/java/expo/modules/interfaces/barcodescanner/BarCodeScannerResult.kt
new file mode 100644
index 0000000..3e60b0e
--- /dev/null
+++ b/node_modules/expo-modules-core/android/src/main/java/expo/modules/interfaces/barcodescanner/BarCodeScannerResult.kt
@@ -0,0 +1,41 @@
+package expo.modules.interfaces.barcodescanner
+
+class BarCodeScannerResult(
+  val type: Int,
+  val value: String?,
+  val raw: String?,
+  var cornerPoints: MutableList<Int>,
+  val height: Int,
+  val width: Int
+) {
+  data class BoundingBox(val x: Int, val y: Int, val width: Int, val height: Int)
+  
+  var referenceImageHeight: Int = height
+  var referenceImageWidth: Int = width
+  
+  val boundingBox: BoundingBox
+    get() {
+      if (cornerPoints.isEmpty()) {
+        return BoundingBox(0, 0, 0, 0)
+      }
+      var minX = Int.MAX_VALUE
+      var minY = Int.MAX_VALUE
+      var maxX = Int.MIN_VALUE
+      var maxY = Int.MIN_VALUE
+
+      var i = 0
+      while (i < cornerPoints.size) {
+        val x: Int = cornerPoints[i]
+        val y: Int = cornerPoints[i + 1]
+
+        minX = minOf(minX, x)
+        minY = minOf(minY, y)
+        maxX = maxOf(maxX, x)
+        maxY = maxOf(maxY, y)
+        i += 2
+      }
+
+      return BoundingBox(minX, minY, maxX - minX, maxY - minY)
+    }
+}
+
diff --git a/node_modules/expo-modules-core/android/src/main/java/expo/modules/interfaces/barcodescanner/BarCodeScannerSettings.kt b/node_modules/expo-modules-core/android/src/main/java/expo/modules/interfaces/barcodescanner/BarCodeScannerSettings.kt
new file mode 100644
index 0000000..08d556b
--- /dev/null
+++ b/node_modules/expo-modules-core/android/src/main/java/expo/modules/interfaces/barcodescanner/BarCodeScannerSettings.kt
@@ -0,0 +1,14 @@
+package expo.modules.interfaces.barcodescanner
+
+class BarCodeScannerSettings {
+  var types: List<Int>? = null
+  
+  fun putTypes(types: List<Int>?) {
+    this.types = types
+  }
+  
+  fun putTypes(types: ArrayList<Double?>?) {
+    this.types = types?.filterNotNull()?.map { it.toInt() }
+  }
+}
+
