diff --git a/node_modules/react-native-webcrypto/android/build.gradle b/node_modules/react-native-webcrypto/android/build.gradle
index e19e09d..8dfd8b9 100644
--- a/node_modules/react-native-webcrypto/android/build.gradle
+++ b/node_modules/react-native-webcrypto/android/build.gradle
@@ -35,5 +35,5 @@ repositories {
 }
 
 dependencies {
-    compile 'com.facebook.react:react-native:+'
+    implementation 'com.facebook.react:react-native:+'
 }
